saleor/order/tests/test_fetch_order_prices_tax_app.py

```diff
@@ -0,0 +1,656 @@
+import json
+from decimal import Decimal
+from unittest.mock import Mock
+
+import pytest
+
+from ...discount import (
+    DiscountType,
+    DiscountValueType,
+    RewardType,
+    RewardValueType,
+    VoucherType,
+)
+from ...graphql.core.utils import to_global_id_or_none
+from ...tax import TaxableObjectDiscountType, TaxCalculationStrategy
+from ...webhook.event_types import WebhookEventSyncType
+from ...webhook.tests.subscription_webhooks.test_create_deliveries_for_taxes import (
+    TAXES_SUBSCRIPTION_QUERY,
+)
+from ...webhook.transport.synchronous.transport import (
+    create_delivery_for_subscription_sync_event,
+)
+from .. import OrderStatus
+from ..calculations import fetch_order_prices_if_expired
+from ..utils import create_order_discount_for_order
+
+
+@pytest.fixture
+def order_with_lines(order_with_lines):
+    order_with_lines.status = OrderStatus.UNCONFIRMED
+    return order_with_lines
+
+
+@pytest.fixture
+def channel_USD(channel_USD):
+    tc = channel_USD.tax_configuration
+    tc.country_exceptions.all().delete()
+    tc.prices_entered_with_tax = False
+    tc.tax_calculation_strategy = TaxCalculationStrategy.TAX_APP
+    tc.tax_app_id = "avatax.app"
+    tc.save()
+    return channel_USD
+
+
+@pytest.fixture
+def subscription_calculate_taxes_for_order(subscription_webhook):
+    return subscription_webhook(
+        TAXES_SUBSCRIPTION_QUERY, WebhookEventSyncType.ORDER_CALCULATE_TAXES
+    )
+
+
+def test_fetch_order_prices_catalogue_promotion(
```

Comment:
We should also have such test for `percentage` discount

Reply:
There is much more missing combination than this one. I created tests to cover updated logic